File: src/Library/ObjectModel/SoundbridgeFactory.vb
Imports System.Net

Public Module SoundbridgeFactory

    Public Function CreateFromTcp(ByVal endPoint As IPEndPoint) As Soundbridge
        Return New Soundbridge(New TcpSoundbridgeClient(endPoint))
    End Function

    Public Function CreateFromTcp(ByVal hostname As String) As Soundbridge
        Return New Soundbridge(New TcpSoundbridgeClient(hostname))
    End Function

    Public Function CreateFromTcp(ByVal hostname As String, ByVal port As Integer) As Soundbridge
        Return New Soundbridge(New TcpSoundbridgeClient(hostname, port))
    End Function

End Module

File: src/Pixa.Soundbridge/ObjectModel/SoundbridgeFactory.cs
using System.Net;

namespace Pixa.Soundbridge.Library
{
    public static class SoundbridgeFactory
    {
        public static Soundbridge CreateFromTcp(IPEndPoint endPoint)
        {
            return new Soundbridge(new TcpSoundbridgeClient(endPoint));
        }

        public static Soundbridge CreateFromTcp(string hostname)
        {
            return new Soundbridge(new TcpSoundbridgeClient(hostname));
        }

        public static Soundbridge CreateFromTcp(string hostname, int port)
        {
            return new Soundbridge(new TcpSoundbridgeClient(hostname, port));
        }
    }
}